### ISSUE TITLE
Help wanted for porting version 2 driver in linux

### ISSUE THREAD

**Issue by Saikatsaha1996** [NewhavenDisplay/FT5X26-Focaltech-Drivers#1]
Hello sir my device chipset in focaltech `ft3658u` was to trying to port my driver to linux mainline.. but I don't have C /C++ skills so I am unable.. I believe if we remove some of code from downstream then it will work in mainline..

I have downstream driver if you really interested to help me.. then just tell me I will share..

Thank you

**Comment by CJoh-NHD** [comment]
Hello,

Here is the example code to using those drivers https://github.com/NewhavenDisplay/FT5X26-Focaltech-Drivers with Linux however the modification must be done on your end.

Regards,

Mateusz Kaczowka  |  Engineering Intern

Newhaven Display International, Inc.
2661 Galvin Court, Elgin, IL 60124 - USA
Call: 847.844.8795  |  Fax: 847.844.8796
www.newhavendisplay.com<https://www.newhavendisplay.com/>

**Comment by Saikatsaha1996** [comment]
> Hello, Here is the example code to using those drivers https://github.com/NewhavenDisplay/FT5X26-Focaltech-Drivers with Linux however the modification must be done on your end. Regards, Mateusz Kaczowka | Engineering Intern Newhaven Display International, Inc. 2661 Galvin Court, Elgin, IL 60124 - USA Call: 847.844.8795 | Fax: 847.844.8796 [www.newhavendisplay.com<https://www.newhavendisplay.com/](http://www.newhavendisplay.com<https://www.newhavendisplay.com/)>

Thank you Sir for your response... Actually my chip is `ft3658u` which is not available..

**Comment by CJoh-NHD** [comment]
Hello,

Can you clarify?

Are you attempting to go from focaltech ft3658u to FT5x26 or from FT5x26 to focaltech ft3658u?
We are unfamiliar with focaltech ft3658u and you will most likely have to reach out to focaltech for information.

Here are the data sheets for FT5x26 (https://support.newhavendisplay.com/hc/en-us/article_attachments/4414394382231/FT5426_5526Application_Note_Ver0.1.pdf) and (https://support.newhavendisplay.com/hc/en-us/article_attachments/4414386815639/FT5x26.pdf)

Modifications of this nature need to be done by the end user unfortunately.

Regards,

**Comment by Saikatsaha1996** [comment]
> Hello, Can you clarify? Are you attempting to go from focaltech ft3658u to FT5x26 or from FT5x26 to focaltech ft3658u? We are unfamiliar with focaltech ft3658u and you will most likely have to reach out to focaltech for information. Here are the data sheets for FT5x26 (https://support.newhavendisplay.com/hc/en-us/article_attachments/4414394382231/FT5426_5526Application_Note_Ver0.1.pdf) and (https://support.newhavendisplay.com/hc/en-us/article_attachments/4414386815639/FT5x26.pdf) Modifications of this nature need to be done by the end user unfortunately. Regards,

Yes My device touchscreen Focaltech "Chip-Id:0x5652" & Chipset is "ft3658u" my Device booted successfully in mainline but unable to use touchscreen..
This is my touchscreen downstrem driver 
https://github.com/Saikatsaha1996/focaltouch-ft3658u

**Comment by CJoh-NHD** [comment]
Hello,

Can you please provide us the part numbers?

Regards,

**Comment by Saikatsaha1996** [comment]
> Hello, Can you please provide us the part numbers? Regards,

I don't know the partnumber.. but i can share if you tell me how can i get ?

This is my driver chip name https://github.com/Saikatsaha1996/android_kernel_oneplus_sm8350/blob/9c86b0cf0172c8ca42ed377af48c0e877ba7a050/arch/arm64/boot/dts/vendor/oplus/porsche/porsche-common.dtsi#L1795

**Comment by CJoh-NHD** [comment]
Hello,

What product are you using exactly if its one of our displays or touch panels there’s a part number on it.

Regards,

**Comment by Saikatsaha1996** [comment]
> Hello, What product are you using exactly if its one of our displays or touch panels there’s a part number on it. Regards,

I am useing Android mobile device name `Realme GT 2` , porting to mainline.. Realme used touchscreen with focaltech... Chipset `FT3658U` and `Chip-Id:0x5652`

**Comment by CJoh-NHD** [comment]
Hello,

We do not produce the Chipset itself they are used in our displays.

For questions regarding the chipset, you should consult the chip manufacturer.

I’m assuming you are modifying the phone to Linux if that’s the case you should be directing inquiry to the phone manufacturer.

Regards,

**Comment by Saikatsaha1996** [comment]
> Hello, We do not produce the Chipset itself they are used in our displays. For questions regarding the chipset, you should consult the chip manufacturer. I’m assuming you are modifying the phone to Linux if that’s the case you should be directing inquiry to the phone manufacturer. Regards,

I was mail them.. in this Mail I'd ( support@focaltech-electronics.com) but bad luck.. mail not sent ..
![Screenshot_20230815-212258](https://github.com/NewhavenDisplay/FT5X26-Focaltech-Drivers/assets/72664192/4e561da8-37b6-4f0d-8ad9-ce6870146a2e)


But okay i  will try tomorrow morning in fst hours to contact with my mobile manufacturer..

**Comment by Saikatsaha1996** [comment]
I was email to my device manufacture but they not reply.. my device chipset ..
![Screenshot_20230817-194258](https://github.com/NewhavenDisplay/FT5X26-Focaltech-Drivers/assets/72664192/8d678e7e-0aff-44ec-920c-8c309512fde3)

**Comment by CJoh-NHD** [comment]
Hello,

Unfortunately, there is nothing we can do on our end.

This is not our product we simply use the chipset, and we have given you all the information we have on it.

Regards
